botocore/handlers.py

```diff
@@ -265,9 +266,9 @@ def _register_for_operations(config, session, service_name):
                          handler, unique_id=unique_id)
 
 
-def signature_overrides(service_data, service_name, session, **kwargs):
-    scoped_config = session.get_scoped_config()
-    service_config = scoped_config.get(service_name)
+def signature_overrides(event_name, config, signature_version, **kwargs):
+    service_name = event_name.split('.')[1]
```

Comment:
Agreed. I've moved this code around a bit due to some other comments, but I also switched it to using `service_model.endpoint_prefix` instead.